.github/workflows/test.yml

```diff
@@ -0,0 +1,26 @@
+name: Test
+
+on:
+  - push
+  - pull_request
+
+jobs:
+    test:
+      runs-on: ubuntu-latest
+
+      steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Test with Gradle
+        run: ./gradlew --no-daemon test
+      - name: Upload artifacts
```

Comment:
We need the test reports if the tests fail. However, if the tests fail, the previous step will fail with a non-zero exit code, which will cause GitHub Actions to stop executing this job. In order to always run the artifact upload step, use an `if` statement in this step. So, this step should look something like:
```yaml
      - if: always()
        uses: actions/upload-artifact@v1
        with:
          name: test-report
          path: build/reports/tests/test/
```